### ISSUE TITLE
[ISSUE] Training on Arabic dataset, unable to attend nor speak properly

### ISSUE THREAD

**Issue by Muhyzater** [NVIDIA/flowtron#105]
I am trying to train a Flowtron on Arabic Speech, however the model is not able to attend and the generated audio is random no matter for how much iterations I train. Given that the quality of audio is fine when discarding the pronunciation.  
The data I have is 25 hours of Arabic speech aligned with text after trimming leading and trailing silences, it is worth mentioning that the same data was successfully used to train a fully functional Tactron 2 model.

I tried training a flowtron model from scratch, and on top of the provided flowtron pre-trained model and another experiment warm-starting from the Tactron 2 model that I have. I also tried the provided English CMUdict and another Arabic pronunciation dictionary.

Now I will leave you with the attention maps after inference and the configurations of the carried experiments.

Attention map after 1.7 M iterations flow = 1 from scratch
![sid1_sigma0 5_attnlayer0](https://user-images.githubusercontent.com/54268896/110484267-2caee600-80f3-11eb-847f-d4ec6ece328a.png)

Attention map after 1.25 M iterations flow = 1 on top of Arabic pre-trained Tacotron 2
![sid1_sigma0 5_attnlayer0](https://user-images.githubusercontent.com/54268896/110484429-510ac280-80f3-11eb-8542-62f4e7011cc1.png)

Attention map after 1.5M iterations flow = 1 on top of English pre-trained flowtron
![sid1_sigma0 5_attnlayer0](https://user-images.githubusercontent.com/54268896/110484514-64b62900-80f3-11eb-83ec-2542228bdafe.png)

The network configuration is the same for the three experiments except for the text dimensions (n_text).

```yml
{
    "train_config": {
        "output_directory": "out_arabic",
        "epochs": 10000000,
        "optim_algo": "RAdam",
        "learning_rate": 1e-5,
        "weight_decay": 1e-6,
        "grad_clip_val": 1,
        "sigma": 1.0,
        "iters_per_checkpoint": 1000,
        "batch_size": 32,
        "seed": 0,
        "checkpoint_path": "",
        "ignore_layers": [],
        "finetune_layers": [],
        "include_layers": ["speaker", "encoder", "embedding"],
        "warmstart_checkpoint_path": "",
        "with_tensorboard": true,
        "fp16_run": false
    },
    "data_config": {
        "training_files": "filelists/train_flowtron.psv",
        "validation_files": "filelists/val_flowtron.psv",
        "text_cleaners": ["flowtron_cleaners"],
        "p_arpabet": 0.5,
        "cmudict_path": "data/Arabic_dict",
        "sampling_rate": 22050,
        "filter_length": 1024,
        "hop_length": 256,
        "win_length": 1024,
        "mel_fmin": 0.0,
        "mel_fmax": 8000.0,
        "max_wav_value": 32768.0,
        "use_attn_prior": false,
        "attn_prior_threshold": 1e-4
    },
    "dist_config": {
        "dist_backend": "nccl",
        "dist_url": "tcp://localhost:54321"
    },
    "model_config": {
        "n_speakers": 1,
        "n_speaker_dim": 128,
        "n_text": 84,
        "n_text_dim": 512,
        "n_flows": 1,
        "n_mel_channels": 80,
        "n_attn_channels": 640,
        "n_hidden": 1024,
        "n_lstm_layers": 2,
        "mel_encoder_n_hidden": 512,
        "n_components": 0,
        "mean_scale": 0.0,
        "fixed_gaussian": true,
        "dummy_speaker_embedding": false,
        "use_gate_layer": true,
        "use_cumm_attention": false
    }
}

```

I cant figure out why it is not attending.
I am personally suspecting it an issue with the pronunciation dictionary, if so, are there any resources for Arabic CMU dictionary.

I can add audio samples if needed.
Thank you so much, your help is appreciated.

**Comment by OmarHory** [comment]
Same issue..
@rafaelvalle

**Comment by rafaelvalle** [comment]
warm-starting from a tacotron 2 model and using the same phoneme dictionary as that tacotron model is a good idea.
remember to lower the learning rate to 5e-4 or 1e-4.

if you're training from scratch, remember to set the attention prior to True: it's set to False in the config you shared.

**Comment by Muhyzater** [comment]
Thank you @rafaelvalle for your answer.
After editing the config as you said, the attention map is starting to look like an attention map, which is a progress for my side.
However, the audio generation is still not improving, and I noticed something and was wondering if you have an answer to.

The number of encoder steps in the attention map looks very small to me (around 8) as in the figure.
![att](https://user-images.githubusercontent.com/54268896/111654931-de47c880-8811-11eb-9b2c-2c9c325d0cfb.png)

Any idea what is causing this issue?

thank you again.

**Comment by rafaelvalle** [comment]
yes, that is surprising.
it says you have an audio sample with 8 characters that lasts 5 seconds.

make sure you trim silences from the beginning and end of your audio files.
make sure the transcription matches the audio file.

**Comment by Muhyzater** [comment]
Thank you @rafaelvalle, 
The issue was in the language.

The cleaner did not recognize the symbols of the Arabic language.
we edit ```symbols.py``` to include the Arabic characters and it looks normal right now.

Hopefully it works out.
Will share with you samples from Tacotron2 and Flowtron in Arabic once ready.

Appreciated.

**Comment by rafaelvalle** [comment]
good to hear.
looking forward to the samples.

**Comment by Muhyzater** [comment]
Thank you @rafaelvalle, The attention map looks great and it is speaking properly with better quality than Tactron.

Some samples in Arabic:
[Tactron2 sample](https://drive.google.com/file/d/1Ljef3H7DIwLCvogBgSNMzcadf5csBU33/view?usp=sharing)
[Flowtron Sample](https://drive.google.com/file/d/1yr9FqdqkCaZMZigR-BdAJ2ts6kA-YXGW/view?usp=sharing)

**Comment by rafaelvalle** [comment]
glad to hear.
